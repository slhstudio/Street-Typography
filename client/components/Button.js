import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';


const StyledButton = styled.button`
  display: flex;
  flex-direction: row;
  font-size: inherit;
  justify-content: center;
  padding: 2rem;
  margin: 1em .5em 1em .5em;
  border: .2rem solid black;
 
  ${props => props.text === 'UPLOAD' && css`
      background: salmon;
      margin: 1.5em 30% 1.5em 30%;
      border: none;
      border-radius: .5em;
  `}

  ${props => props.text === 'Google' && css`
      background: none;
      margin: 1.5em 30% 1.5em 30%;
      border: none;
      border-radius: .5em;
  `}
`;

const Button = (props) => {
  return (
    <StyledButton 
      type={props.type}
      onClick={props.onClick}
      text={props.text}
    >
      {props.text}
    </StyledButton>
  )
}

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired
}

export default Button;

{/* <span class="label">Sign in with:</span>
    <div id="customBtn" class="customGPlusSignIn">
      <span class="icon"></span>
      <span class="buttonText">Google</span>
    </div> */}