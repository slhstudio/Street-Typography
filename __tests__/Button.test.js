import React from 'react';
import Button from '../client/components/Button';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components'

const fakeProps = {
  type : 'submit',
  onClick : jest.fn(),
  text: 'UPLOAD'
}

describe('<Button/>', () => {
  it ('renders and displays properly', () => {
    const wrapper = shallow(<Button {...fakeProps}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({text: 'SAVE'});
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});