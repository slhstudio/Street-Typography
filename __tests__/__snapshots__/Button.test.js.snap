// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders and displays properly 1`] = `
<styled.button
  onClick={[MockFunction]}
  type="submit"
>
  UPLOAD
</styled.button>
`;

exports[`<Button/> renders and displays properly 2`] = `
<styled.button
  onClick={[MockFunction]}
  type="submit"
>
  SAVE
</styled.button>
`;
